src/LoggerServiceFactory.php

```diff
@@ -34,9 +44,108 @@ public function __invoke(ContainerInterface $container, $name, ?array $options =
      * Proxies to `__invoke()`.
      *
      * @return Logger
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
      */
     public function createService(ServiceLocatorInterface $serviceLocator)
     {
         return $this($serviceLocator, Logger::class);
     }
+
+    /**
+     * Process and return the configuration from the container.
+     *
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
+     */
+    protected function processConfig(array &$config, ContainerInterface $services)
+    {
+        if (
+            isset($config['writer_plugin_manager'])
+            && is_string($config['writer_plugin_manager'])
+            && $services->has($config['writer_plugin_manager'])
+        ) {
+            $config['writer_plugin_manager'] = $services->get($config['writer_plugin_manager']);
+        }
+
+        if (
+            (! isset($config['writer_plugin_manager'])
+                || ! $config['writer_plugin_manager'] instanceof AbstractPluginManager)
+            && $services->has('LogWriterManager')
+        ) {
+            $config['writer_plugin_manager'] = $services->get('LogWriterManager');
+        }
+
+        if (
+            isset($config['processor_plugin_manager'])
+            && is_string($config['processor_plugin_manager'])
+            && $services->has($config['processor_plugin_manager'])
+        ) {
+            $config['processor_plugin_manager'] = $services->get($config['processor_plugin_manager']);
+        }
+
+        if (
+            (! isset($config['processor_plugin_manager'])
+                || ! $config['processor_plugin_manager'] instanceof AbstractPluginManager)
+            && $services->has('LogProcessorManager')
+        ) {
+            $config['processor_plugin_manager'] = $services->get('LogProcessorManager');
+        }
+
+        if (! isset($config['writers'])) {
```

Comment:
Based on how this is consumed in line 99, this should also include `|| ! is_iterable($config['writers'])` in the conditional to ensure we can actually traverse it.

Reply:
I had to add this after the isset `if` aswell, due to the Logger itself only supporting real arrays
https://github.com/laminas/laminas-log/blob/2.15.x/src/Logger.php#L209

```php
        if (! is_array($config['writers'])) {
            $config['writers'] = iterator_to_array($config['writers']);
        }
```